civihr_employee_portal/updates/7044.php

```diff
@@ -12,27 +12,40 @@ function civihr_employee_portal_update_7044() {
 
   $addresses = _get_personal_address_count();
   foreach ($addresses as $address) {
-    if ($address['total'] != 2) {
+    if ($address['total'] < 2) {
       continue;
     }
 
     $result = civicrm_api3('Address', 'get', [
       'location_type_id' => 'Personal',
       'contact_id' => $address['contact_id']
     ]);
-    $processed = 0;
-    foreach ($result['values'] as $contactAddress) {
-      $updateAnyAddress = $contactAddress['is_primary'] == 0 && $address['primary_count'] == 1;
-      $updateSecondAddress = $address['primary_count'] != 1 && $processed == 1;
-
-      if ($updateAnyAddress || $updateSecondAddress) {
-        $contactAddress['location_type_id'] = $locationType['id'];
-        $contactAddress['is_primary'] = 0;
-        civicrm_api3('Address', 'create', $contactAddress);
-      }
-
-      $processed++;
+
+    $addresses = array_values($result['values']);
+    $primaryIndex = array_search('1', array_column($addresses, 'is_primary'));
+    if ($primaryIndex === FALSE) {
+      $addresses[0]['is_primary'] = 1;
+      civicrm_api3('Address', 'create', $addresses[0]);
+      $primaryIndex = 0;
     }
+
+    unset($addresses[$primaryIndex]);
+    _update_address_to_works($addresses, $locationType['id']);
```

Comment:
I might be missing something, but if I understand  the code correctly, l think all contact addressed will be changed to work address type like this even the primary personal  address

Reply:
Exactly, the ticket indicates not more that one personal address. I'm not sure (but will confirm) if work address should also be restricted to one. In that case, we may have to unset the address type for the ones changed to `Work` type.